Classes/EventStore/EventStream.php

```diff
@@ -11,100 +11,93 @@
  * source code.
  */
 
-use Neos\Cqrs\Event\EventInterface;
-use Neos\Cqrs\Event\EventTransport;
+use Neos\Cqrs\Event\EventMetadata;
+use Neos\Cqrs\Event\EventWithMetadata;
+use Neos\Cqrs\Event\EventTypeResolver;
 use TYPO3\Flow\Annotations as Flow;
+use TYPO3\Flow\Property\PropertyMapper;
+use TYPO3\Flow\Property\PropertyMappingConfiguration;
 
 /**
  * EventStream
  */
-class EventStream implements \IteratorAggregate
+final class EventStream implements \Iterator
```

Comment:
The `EventStream` is now a wrapper for any `\Iterator` for `EventFromStream` instances (for example `DoctrineStreamIterator` below).
It contains a getter for the stream *version* and *lazily* converts every event to an instance of `EventWithMetadata` in `current()`.